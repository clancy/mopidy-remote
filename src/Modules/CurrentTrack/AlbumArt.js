/* @flow */
import React, { Component, PropTypes } from 'react';
import {
  View,
  StyleSheet,
  Text,
  Image
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const AlbumArt = (props) => (
  <View style={[styles.container, !props.albumArtUri ? styles.iconContainer : null]}>
    {(props.albumArtUri
      ? <Image style={styles.albumArtImage} source={{uri: props.albumArtUri }}/>
      : <Icon name='error-outline' style={styles.icon} />
    )}
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 3,
    backgroundColor: 'black'
  },
  iconContainer: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  albumArtImage: {
    flex:1
  },
  icon: {
    color: 'white',
    fontSize: 200
  }
});

export default AlbumArt;
