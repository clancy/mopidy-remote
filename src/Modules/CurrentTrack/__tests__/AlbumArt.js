import 'react-native';
import React from 'react';
import AlbumArt from '../AlbumArt';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('AlbumArt tests', () => {
  test('renders icon without albumArtUri', () => {
    const tree = renderer.create(
      <AlbumArt />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders image with albumArtUri', () => {
    const tree = renderer.create(
      <AlbumArt albumArtUri={'http://someimage.url'}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
