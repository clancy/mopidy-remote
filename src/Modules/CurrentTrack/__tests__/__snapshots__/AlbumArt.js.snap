exports[`AlbumArt tests renders icon with albumArtUri 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flex": 3,
      },
      null,
    ]
  }>
  <Image
    source={
      Object {
        "uri": "http://someimage.url",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    } />
</View>
`;

exports[`AlbumArt tests renders icon without albumArtUri 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flex": 3,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "white",
          "fontSize": 200,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }>
    
  </Text>
</View>
`;
